### PR TITLE
Add AMQP failover endpoint support

### PR DESCRIPTION
**Pull request by HeshanSudarshana**
### Purpose
This PR adds the AMQP failover endpoint support. `controlPlane.eventHub.jmsConnectionParameters` config in the config.toml should be provided as an array as below.
```
[controlPlane.eventHub.jmsConnectionParameters]
eventListeningEndpoints = ["amqp://admin:admin@172.17.0.1:5672?retries='5'&connectdelay='30000'","amqp://admin:admin@localhost:5672?retries='3'&connectdelay='30000'"]
```

### Issues
Fixes https://github.com/wso2/product-microgateway/issues/1662

Related PR - https://github.com/wso2/product-microgateway/pull/1629
Related changes - https://github.com/wso2/product-microgateway/blob/72bf6c844ebfc0355165887892f6b6f89a07705f/adapter/pkg/messaging/mainListener.go, https://github.com/wso2/product-microgateway/blob/72bf6c844ebfc0355165887892f6b6f89a07705f/adapter/pkg/messaging/mainListener.go

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by HiranyaKavishani — APPROVED** [review]

File: adapter/internal/messaging/connection.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
+package messaging
```
Comment (praminda) [inline adapter/internal/messaging/connection.go:1]: License missing

**Comment by praminda** [comment]
Please add the PR template again.

File: adapter/internal/messaging/connection.go (hunk 1 above)
Comment (HeshanSudarshana) [reply adapter/internal/messaging/connection.go:19]: Added with https://github.com/wso2/product-microgateway/pull/1664/commits/c4d33b057476a7497b514e9782443cc17ea76823

**Comment by HeshanSudarshana** [comment]
> Please add the PR template again.

Added with https://github.com/wso2/product-microgateway/pull/1664/commits/c4d33b057476a7497b514e9782443cc17ea76823

**Review by praminda — APPROVED** [review]

**Pull request merged**